File: src/java/day04/Gril.java
package main.java.day04;

public class Gril implements Comparable<Gril> {
    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public int getFaceValue() {
        return faceValue;
    }

    public void setFaceValue(int faceValue) {
        this.faceValue = faceValue;
    }

    private String name;
    private int faceValue;

    @Override
    public int compareTo(Gril o) {
        return o.getFaceValue() - this.getFaceValue();
    }
}

File: src/scala/day04/Gril.scala
package main.scala.day04

class Gril(val name:String, var faceValue:Int)  extends Ordered[Gril]{
  override def compare(that: Gril): Int = {
    this.faceValue - that.faceValue
  }

  override def toString: String = s"Gril($name, $faceValue)"
}

class MyGril[T] {
  def faceValueSome(g1:Gril, g2:Gril):Gril={
    if(g1 > g2) g1 else g2
  }
}

object Gril extends App {
  var arr = Array[Gril](new Gril("苍老师", 90), new Gril("波多老师", 95), new Gril("吉泽老师", 94))

  val sorted =  arr.sortBy(x=> x)
  println(sorted.toBuffer)

  val myGril = new MyGril[Gril]
  val s = myGril.faceValueSome(new Gril("苍老师", 90), new Gril("波多老师", 95) )
  println(s.name)
}